lib/livebook_web/live/settings_live.ex

```diff
@@ -132,6 +192,42 @@ defmodule LivebookWeb.SettingsLive do
   def handle_params(_params, _url, socket), do: {:noreply, socket}
 
   @impl true
+  def handle_event("cancel_autosave_path", %{}, socket) do
+    {:noreply,
+     update(
+       socket,
+       :autosave_path_state,
+       &%{&1 | dialog_opened?: false, file: file_for_local_path()}
+     )}
+  end
+
+  def handle_event("set_autosave_path", %{}, socket) do
+    path = socket.assigns.autosave_path_state.file.path
+
+    Livebook.Settings.set_autosave_path(path)
+
+    {:noreply,
+     update(
+       socket,
+       :autosave_path_state,
+       &%{&1 | dialog_opened?: false, file: file_for_local_path()}
```

Comment:
I think it will be cleaner if we just have `defp autosave_dir` and `defp default_autosave_dir` that return the corresponding file struct.